src/main/java/com/atlassian/db/replica/internal/aurora/AuroraClusterDiscovery.java

```diff
@@ -11,17 +12,21 @@
 import static java.util.stream.Collectors.toList;
 
 public final class AuroraClusterDiscovery {
-    private final AuroraConnectionDetails auroraConnectionDetails;
+    private final ConnectionPoolPerReplicaProvider connectionPoolPerReplicaProvider;
 
-    public AuroraClusterDiscovery(AuroraConnectionDetails auroraConnectionDetails) {
-        this.auroraConnectionDetails = auroraConnectionDetails;
+    public AuroraClusterDiscovery(ConnectionPoolPerReplicaProvider connectionPoolPerReplicaProvider) {
+        this.connectionPoolPerReplicaProvider = connectionPoolPerReplicaProvider;
     }
 
     public Collection<Database> getReplicas(Supplier<Connection> connectionSupplier) {
         try (final Connection connection = connectionSupplier.get()) {
             final AuroraReplicasDiscoverer discoverer = createDiscoverer(connection);
             return discoverer.fetchReplicasUrls(connection).stream()
-                .map(auroraUrl -> new AuroraReplicaNode(auroraUrl, auroraConnectionDetails))
+                .map(auroraUrl -> {
+                    String id = auroraUrl.getEndpoint().getServerId();
+                    ConnectionPool connectionPool = connectionPoolPerReplicaProvider.getConnectionPool(auroraUrl);
+                    return new AuroraReplicaNode(id, connectionPool);
```

Comment:
What is the responsibility of AuroraReplicaNode now? How is it different than ConnectionPool